native/core/src/parquet/parquet_exec.rs

```diff
@@ -0,0 +1,261 @@
+    match datafusion_comet_objectstore_hdfs::object_store::hdfs::HadoopFileSystem::new(url.as_ref())
+    {
+        Some(object_store) => {
+            let path = object_store.get_path(url.as_str());
+            Ok((Box::new(object_store), path))
+        }
+        _ => {
+            return Err(object_store::Error::Generic {
+                store: "HadoopFileSystem",
+                source: Box::new(datafusion_comet_objectstore_hdfs::object_store::hdfs::HadoopFileSystem::HdfsErr::Generic(
+                    "Could not create hdfs object store".to_string(),
+                )),
+            });
+        }
+    }
+}
+#[cfg(not(feature = "hdfs"))]
+fn parse_hdfs_url(_url: &Url) -> Result<(Box<dyn ObjectStore>, Path), object_store::Error> {
+    Err(object_store::Error::Generic {
+        store: "HadoopFileSystem",
+        source: "Hdfs support is not enabled in this build".into(),
+    })
+}
+
+// parses the url, registers the object store, and returns a tuple of the object store url and object store path
+pub(crate) fn prepare_object_store(
+    runtime_env: Arc<RuntimeEnv>,
+    url: String,
+) -> Result<(ObjectStoreUrl, Path), ExecutionError> {
+    let mut url = Url::parse(url.as_str()).unwrap();
+    let mut scheme = url.scheme();
+    if scheme == "s3a" {
+        scheme = "s3";
+        url.set_scheme("s3")
+            .expect("Could not convert scheme from s3a to s3");
+    }
+    let url_key = format!(
+        "{}://{}",
+        scheme,
+        &url[url::Position::BeforeHost..url::Position::AfterPort],
+    );
+
+    let (object_store, object_store_path): (Box<dyn ObjectStore>, Path) = if scheme == "hdfs" {
+        match parse_hdfs_url(&url) {
+            Ok(r) => r,
+            Err(e) => return Err(ExecutionError::GeneralError(e.to_string())),
+        }
+    } else {
+        match parse_url(&url) {
+            Ok(r) => r,
+            Err(e) => return Err(ExecutionError::GeneralError(e.to_string())),
+        }
+    };
+    let object_store_url = ObjectStoreUrl::parse(url_key.clone())?;
+    // runtime_env.register_object_store(object_store_url.as_ref(), Arc::from(object_store));
+    runtime_env.register_object_store(&url, Arc::from(object_store));
+    Ok((object_store_url, object_store_path))
+}
+
+pub(crate) fn get_file_groups_single_file(
+    path: &Path,
+    file_size: u64,
+    start: i64,
+    length: i64,
+) -> Vec<Vec<PartitionedFile>> {
+    let mut partitioned_file = PartitionedFile::new_with_range(
+        String::new(), // Dummy file path. We will override this with our path so that url encoding does not occur
+        file_size,
+        start,
+        start + length,
+    );
+    partitioned_file.object_meta.location = (*path).clone();
+    vec![vec![partitioned_file]]
+}
+
+fn get_options(session_timezone: &str) -> (TableParquetOptions, SparkParquetOptions) {
+    let mut table_parquet_options = TableParquetOptions::new();
+    // TODO: Maybe these are configs?
```

Comment:
do you mean to obtain those values from external config like spark config?

Reply:
Yeah that's my TODO that Parth relocated. I'm not sure what the scenario is when you _wouldn't_ want filters pushed down to scans, but I wanted to have a conversation about it.